src/Resources/contao/library/Contao/StringUtil.php

```diff
@@ -547,8 +580,21 @@ public static function parseSimpleTokens($strString, $arrData)
 
 		// Replace tokens
 		$strReturn = str_replace('?><br />', '?>', $strReturn);
-		$strReturn = preg_replace('/##([A-Za-z0-9_]+)##/i', '<?= $arrData[\'$1\'] ?>', $strReturn);
-		$strReturn = str_replace("] ?>\n", '] . "\n" ?>' . "\n", $strReturn); // see #7178
+		$strReturn = preg_replace_callback(
+			'/##([^=!<>\s]+)##/',
```

Comment:
The `+` should be ungreedy in this case so that things like `##token1##,##token1##` get matched as two seperate tokens: `/##([^=!<>\s]+?)##/`